src/views/iam/ImportRole.vue

```diff
@@ -0,0 +1,267 @@
+    },
+    handleSubmit (e) {
+      e.preventDefault()
+      this.form.validateFields((err, values) => {
+        if (err) {
+          return
+        }
+        const params = {}
+        for (const key in values) {
+          const input = values[key]
+          if (input === undefined) {
+            continue
+          }
+          if (key === 'file') {
+            continue
+          }
+
+          params[key] = input
+        }
+
+        if (this.fileList.length !== 1) {
+          return
+        }
+
+        this.loadCSVFile(this.fileList[0]).then((rules) => {
+          rules.forEach(function (values, index) {
+            for (const key in values) {
+              params['rules[' + index + '].' + key] = values[key]
+            }
+          })
+
+          this.importRole(params)
+        })
+      })
+    },
+    closeAction () {
+      this.$emit('close-action')
+    },
+    importRole (params) {
+      this.loading = true
+      api('importRole', {}, 'POST', params).then(json => {
+        const role = json.importroleresponse.role
+        if (role) {
+          this.$emit('refresh-data')
+          this.$notification.success({
+            message: 'Import Role',
+            description: 'Sucessfully imported Role ' + params.name
+          })
+        }
+      }).catch(error => {
+        this.$notifyError(error)
+      }).finally(() => {
+        this.loading = false
+        this.closeAction()
+      })
+    },
+    csvToJson (csv) {
+      var lines = csv.split('\n')
+      var result = []
+      var headers = lines[0].split(',')
+
+      lines.map((line, indexLine) => {
+        if (indexLine < 1) return // Jump header line
+        var obj = {}
+        var currentline = line.split(',')
+
+        headers.map((header, indexHeader) => {
+          obj[header] = currentline[indexHeader]
+        })
+        result.push(obj)
+      })
+      result.pop() // remove the last item because undefined values
+
+      return result
+    },
+    loadCSVFile (file) {
+      return new Promise((resolve, reject) => {
+        if (window.FileReader) {
+          var reader = new FileReader()
+          reader.onload = (event) => {
+            var csv = event.target.result
+            // var lines = csv.split('\n')
+            // var result = []
+            // var headers = lines[0].split(',')
+
+            // lines.map((line, indexLine) => {
+            //  if (indexLine < 1) return
+            //  var obj = {}
+            //  var currentline = line.split(',')
+
+            //  headers.map((header, indexHeader) => {
+            //    obj[header] = currentline[indexHeader]
+            //  })
+            //  result.push(obj)
+            // })
+            // result.pop()
+
+            var rules = this.csvToJson(csv)
+            resolve(rules)
+          }
+
+          reader.onerror = (event) => {
+            if (event.target.error.name === 'NotReadableError') {
+              alert('Cannot read file!')
```

Comment:
Probably trying to show validation error but we don't show `alert`.
Not sure if this can help https://github.com/apache/cloudstack-primate/blob/5ce6d1ca4606add910dd9bd40a410d57306bec78/src/views/offering/AddComputeOffering.vue#L497-L502

Reply:
> Probably trying to show validation error but we don't show `alert`.
> Not sure if this can help
> 
> https://github.com/apache/cloudstack-primate/blob/5ce6d1ca4606add910dd9bd40a410d57306bec78/src/views/offering/AddComputeOffering.vue#L497-L502

@shwstppr Used this.$error() for alert dialog